### DISCUSSION TITLE
Using multiple `crossref` options

### DISCUSSION THREAD

**Discussion by lorenzoFabbri** [quarto-dev/quarto-cli#7653] [Q&A]
### Description

In my main Quarto document I am including another one using `{{ < include ... > }}`. In the main document I am using the default options for `crossref`, while in the second one (the one included), I specified the following:

```
crossref:
  fig-title: "Supplementary Figure"
  fig-prefix: "Supplementary Figure"
  tbl-title: "Supplementary Table"
  tbl-prefix: "Supplementary Table"
```

The problem is that now all the tables and figures are referenced using this options (e.g., there are no more *normal* Table X). Is it possible to combine multiple options for `crossref` in different documents?

Moreover, when I cite inline for instance a table using `@`, there is no space between the last word before the reference and the reference itslef (e.g., "given inSupplementary Table X").

I am using R version 4.3.1 (2023-06-16) and Quarto version 1.4.487.

**Comment by cderv** [comment]
> In the main document I am using the default options for crossref, while in the second one (the one included), I specified the following

In our doc (https://quarto.org/docs/authoring/includes.html) we say in a Red important callout 

> Include shortcodes are equivalent to copying and pasting the text from the included file into the main file. This means that relative references (links, images, etc.) inside the included file resolve based on the directory of the main file not the included file.
>
> It also means that **if the included file has a metadata block, that block will take effect in all included files**. In most cases, having metadata blocks in an included file will cause unexpected behavior.

What exactly would you like to do ? How to you expect "include" to behave ? 

As we explain in the doc it is "equivalent to copying and pasting the text from the included file into the main file" - It is not including another independently rendered document

**Reply by lorenzoFabbri** [reply]
Yes I know that... I wanted to cross reference both *normal* tables and tables coming from the supplementary material. They come from two different Quarto documents, and scientific journals expect the latter to be referenced differently (e.g., Supplementary Table 1).

**Reply by mcanouil** [reply]
Use 1.4 and the new cross-reference system to define custom cross-reference.
See <https://quarto.org/docs/prerelease/1.4/crossref.html>.

(You'll never be able to use include the way you want.)

**Reply by lorenzoFabbri** [reply]
Thank you. Although I doubt people will actually use it since supplementary material is basically only included as a separate file.

**Reply by mcanouil** [reply]
> Although I doubt people will actually use it

Why is that?
You can't cross-reference something that isn't there (note that this is also true in pure Word, or it is just regular text, not a cross-reference), so in order to cross-reference supplementary they have to be there in the first place.
Note that there are various ways to "remove" the supplementary content later on.

**Reply by lorenzoFabbri** [reply]
Yes I know of course. But the Supplementary Material (SM) is a separate document. It would be nice to be able to cross reference using e.g., the full *path*, like "see sm.qmd#tbl-desc1" without that table actually being displayed in the rendered document...

You mean you would open the rendered document and delete what you do not want? Thank you for the answers btw.

**Reply by cderv** [reply]
> It would be nice to be able to cross reference using e.g., the full path, like "see sm.qmd#tbl-desc1" without that table actually being displayed in the rendered document...

This would be indeed a nice feature request for such thing as Supplementary Materials; Currently, maybe mixing the custom crossref labels (https://quarto.org/docs/prerelease/1.4/crossref.html#example-supplemental-figures-and-tables) with include of the supplementary document can help. You will get one output only though. 

There are some work to do in Quarto before even being able to think about including such feature. @cscheid do you think this is something that we could make work in the feature ? I understand the need for academic content and supplementary documents.


Also @lorenzoFabbri 
Are you aware of the work on Manuscript Project type ? https://quarto.org/docs/manuscripts/
Don't think it answers the need here, but still interesting to know for academic usage.

**Reply by lorenzoFabbri** [reply]
@mcanouil I am trying to use the custom cross-reference... In the main Quarto document (`main.qmd`) I am defining the following:

```
crossref:
  custom: 
    - kind: float
      ref-type: supptbl
      latex-env: supptbl
      prefix: Table S
      name: Table S
      space-before-numbering: false
    - kind: float
      ref-type: suppfig
      latex-env: suppfig
      prefix: Figure S
      name: Figure S
      space-before-numbering: false
```

In `main.qmd` I am then including `{{< include _si.qmd >}}`. `_si.qmd` contains the following:

````
```{r info-chems}
#| label: supptbl-info-chems
#| tbl-cap: Information about non-persistent EDCs

edcs_info
```
````

When I render the document, I get the following warning, though: `WARNING (/opt/quarto/share/filters/main.lua:11222) Unable to resolve crossref @supptbl-info-chems`. Indeed, in the render `docx`, there is no caption for this table. I did not find any worked example on using this new feature.

**Reply by mcanouil** [reply]
Please provide a complete and small (not) working reproducible example (document or Git repository), thank you.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by lorenzoFabbri** [reply]
Main Quarto document:

````qmd
---
title: "Estimating associations..."

format:
  docx:
    reference-doc: reference_document.docx
    citeproc: false

crossref:
  custom: 
    - kind: float
      ref-type: supptbl
      latex-env: supptbl
      prefix: Table S
      name: Table S
      space-before-numbering: false
    - kind: float
      ref-type: suppfig
      latex-env: suppfig
      prefix: Figure S
      name: Figure S
      space-before-numbering: false

link-citations: true
linkcolor: blue

editor_options: 
  chunk_output_type: console

execute:
  echo: false
  warning: false
  error: false
  message: false
---

```{r}
here::i_am("output/paper/manuscript.qmd")
```

{{< include _si.qmd >}}

````

`_si.qmd` contains:

````qmd
```{r info-chems}
#| label: supptbl-info-chems
#| tbl-cap: Information about non-persistent EDCs

edcs_info
```
````

[reference_document.docx](https://github.com/quarto-dev/quarto-cli/files/13428520/reference_document.docx)

**Reply by cderv** [reply]
Could you make the example reproducible ? `edcs_info` or `output/paper/manuscript.qmd` is not available to us.

**Reply by cderv** [reply]
I think you hit something we don't support yet - teach **knitr** to forward some custom crossref element. The specific label prefix is unknown to **knitr** and not passed currently to the .md output.  It should probably be pass through... 

Try the usual Markdown syntax, without trying to set some cell options

````markdown
---
title: "Estimating associations..."
format: html
keep-md: true
crossref:
  custom: 
    - kind: float
      ref-type: supptbl
      latex-env: supptbl
      prefix: Table S
      name: Table S
      space-before-numbering: false
    - kind: float
      ref-type: suppfig
      latex-env: suppfig
      prefix: Figure S
      name: Figure S
      space-before-numbering: false
link-citations: true
linkcolor: blue
editor_options: 
  chunk_output_type: console
execute:
  echo: false
  warning: false
  error: false
  message: false
---

{{< include _si.qmd >}}

See @supptbl-info-chems
````

With `_si.qmd` being
````markdown

::: {#supptbl-info-chems}

```{r info-chems}
knitr::kable(head(cars))
```

Information about non-persistent EDCs
:::


````

It should works 
![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/293a1348-5d73-40f8-99a4-a01d42e1a383)


(_and see that I needed to build a reproducible to help - that is why it is really helpful and important that one is provided when reporting issue or asking question - it really saves us time and help us help you!_)

**Reply by lorenzoFabbri** [reply]
You're absolutely right. I would have modified it to be completely reproducible soon.
The thing is that in my document that was a `gt` table... I see what you did. Wrapping the R code. I'll test it.

**Reply by cderv** [reply]
I opened an issue about the new custom crossref  and cell engine not working as users are trying 
* https://github.com/quarto-dev/quarto-cli/issues/7665
